src/App.tsx

```diff
@@ -1,25 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
-
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const totalItems = 42; // Total number of items
+  // const totalPages = Math.ceil(totalItems / perPage);
+
+  const items = getNumbers(
+    (currentPage - 1) * perPage + 1,
+    Math.min(currentPage * perPage, totalItems),
+  ).map(n => `Item ${n}`);
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(parseInt(e.target.value, 10));
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {(currentPage - 1) * perPage + 1} -{' '}
+        {Math.min(currentPage * perPage, totalItems)} of {totalItems})
```

Comment:
pls move complex expressions/calculations to separate consts, don't use them in jsx
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**